### ISSUE TITLE
FormattedRelative doesn't update state.now on props.value change.

### ISSUE THREAD

**Issue by iggyfisk** [yahoo/react-intl#495]
I'm using react-intl for an app with lots of live data. FormattedRelative is used as a "Last change for this item" marker. When that timestamp changes, FormattedRelative will sometimes display things like "in 27 seconds" as if the last change happened in the future. This of course is because the comparison inside the component is between `this.state.now` and `this.props.value`. The props change happens in between the timer ticks to update `this.state.now`. After the next tick the output is correct again. Check this [fiddle](http://jsfiddle.net/ybm69j1z/2/), click the button and you'll see "in 5 seconds" etc.

One idea is to update `this.state.now` when the `value` prop changes, something like
```javascript
function componentWillReceiveProps(next) {
	if (this.props.value && this.props.value.getTime() != next.value.getTime()) {
		this.setState({now: this.context.intl.now()})
	}
}
```
But you probably have better ideas on how to handle this, or if it shouldn't be handled at all and I should restructure my app somehow instead.

**Comment by ericf** [comment]
@iggyfisk clearly a bug, thanks for reporting it. What you're suggesting makes sense to fix it. I'll get a PR opened and a patch release out soon…

**Comment by ptomasroos** [comment]
Thanks @ericf thats brilliant!

**Comment by ericf** [comment]
Fix released in `v2.1.3`.

**Comment by ptomasroos** [comment]
Big thanks for the support @ericf ! ✋
